src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -205,4 +212,48 @@ public Coords getComputerFireCoords() {
 
         return new Coords(across, down);
     }
+
+    public void placeAIShip(String name){
+        String orient;
+        int dir;
+        boolean fail;
+        Random randNum = new Random();
+
+
+        do {
+            int x = (randNum.nextInt(GRID_SIZE) + 1);
+            int y = (randNum.nextInt(GRID_SIZE) + 1);
+
+            dir = (randNum.nextInt(1) + 0);
+
+            if (dir == 0) orient = "horizontal";
+            else if (dir == 1) orient = "vertical";
+            else orient = "well fuck";
+
+            //INCLUDE FUNCTION TO FIND OUT IF OVERLAP OR OUT OF BOUNDS
```

Comment:
Once I get some bugs fixed any my next PR merged (tomorrow), updateShipPosition will automatically disallow overlap. You won't need to test it with a separate method with the fail var, just so ya know. It'll make more sense when I finish my code for it.